evalml/automl/automl_search.py

```diff
@@ -266,11 +272,21 @@ def __init__(self,
         self.search_iteration_plot = None
         self._interrupted = False
 
+        self._prepended_components = []
+        self._prepended_components_parameters = {}
+        if data_check_actions is not None:
+            self.data_check_actions = data_check_actions
+            _prepended_components_and_parameters = _make_component_list_from_actions(self.data_check_actions)
+            for component, parameters in _prepended_components_and_parameters:
+                self._prepended_components_parameters.update({component.name: parameters})
```

Comment:
This will be problematic if multiple of the same components, but for now, combining components since AutoML isn't great at handling multiple of the same components either 😬  (https://github.com/alteryx/evalml/issues/2049)